### ISSUE TITLE
Continious crash: java.lang.NoClassDefFoundError: Not a primitive type: '\u0000'

### ISSUE THREAD

**Issue by dadosch** [iNPUTmice/Conversations#3691]
#### General information

* **Version:** 2.7.1+fcr
* **Device:** Samsung Galaxy A3 2016
* **Android Version:**  Android 7.0 (stock)
* **Server name:** self hosted
* **Server software:** prosody
* **Installed server modules:** Stream Managment, CSI, MAM
* **Conversations source:** F-Droid

#### Steps to reproduce

1. open conversations

#### Expected result
Conversations should open regularly

#### Actual result
Conversations crashes at start, even cleaning cache doesn't solve it.

#### Debug output

````
05-03 00:28:43.008 15365 15365 D conversations: ForegroundService: on
05-03 00:28:43.009 15365 15365 D conversations: update unread count to 0
05-03 00:28:43.060 15365 15388 D conversations: restoring messages...
05-03 00:28:43.073 15365 15365 D conversations: ForegroundService: on
05-03 00:28:43.097 15365 15394 D conversations: account@domain.tld: connecting
05-03 00:28:43.113 15365 15391 D conversations: start merging phone contacts with roster
05-03 00:28:43.113 15365 15391 D conversations: finished merging phone contacts
05-03 00:28:43.131 15365 15365 D conversations: resetting all attempt counts
05-03 00:28:47.541 15365 15394 D conversations: Resolver: [Result{ip='8.8.8.8', hostame='domain.tld', port=5222, directTls=false, authenticated=false, priority=5}, Result{ip='ipv6ip', hostame='domain.tld', port=5222, directTls=false, authenticated=false, priority=5}]
05-03 00:28:47.543 15365 15394 D conversations: verified hostname null
05-03 00:28:47.544 15365 15394 D conversations: account@domain.tld: using values from resolver domain.tld/8.8.8.8:5222 tls: false
05-03 00:28:47.931 15365 15394 D XmppDomainVerifier: searching for domain.tld in srvNames: [] xmppAddrs: [] domains:[*.domain.tld, domain.tld]
05-03 00:28:47.931 15365 15394 D XmppDomainVerifier: comparing .de and .domain.tld
05-03 00:28:47.931 15365 15394 D XmppDomainVerifier: domain domain.tld matched domain.tld
05-03 00:28:47.934 15365 15394 D conversations: account@domain.tld: TLS connection established
05-03 00:28:48.073 15365 15394 D conversations: account@domain.tld: protocol=TLSv1.3 cipher=TLS_AES_256_GCM_SHA384
05-03 00:28:48.075 15365 15394 D conversations: account@domain.tld/Conversations.2zwI: Authenticating with PLAIN
05-03 00:28:48.132 15365 15394 D conversations: account@domain.tld: logged in
05-03 00:28:48.246 15365 15394 D conversations: account@domain.tld: not counting stanza(IqPacket). Not in smacks session.
05-03 00:28:48.249 15365 15394 D conversations: account@domain.tld: starting service discovery
05-03 00:28:48.271 15365 15394 D conversations: account@domain.tld: server caps came from cache
05-03 00:28:48.288 15365 15394 D conversations: account@domain.tld: stream management(3) enabled (resumable)
05-03 00:28:48.331 15365 15394 D conversations: account@domain.tld: Requesting block list
05-03 00:28:48.405 15365 15394 D conversations: Received blocklist update from server
05-03 00:28:48.495 15365 15394 E AndroidRuntime: FATAL EXCEPTION: Thread-6
05-03 00:28:48.495 15365 15394 E AndroidRuntime: Process: eu.siacs.conversations, PID: 15365
05-03 00:28:48.495 15365 15394 E AndroidRuntime: java.lang.NoClassDefFoundError: Not a primitive type: '\u0000'
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.parser.IqParser.bundle(IqParser.java:270)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.crypto.axolotl.AxolotlService$4.onIqPacketReceived(AxolotlService.java:552)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processIq(XmppConnection.java:763)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:636)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:484)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.switchOverToTls(XmppConnection.java:807)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:468)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.connect(XmppConnection.java:376)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at eu.siacs.conversations.xmpp.XmppConnection.run(XmppConnection.java:455)
05-03 00:28:48.495 15365 15394 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:762)
````

**Comment by dadosch** [comment]
The same thing hapens, when starting with no connectivity and disabling DNSSEC check:
After a restart with connctivity, it still failes with the error in the title

**Comment by iNPUTmice** [comment]
Looks like a combination of corrupted data on your Prosody and a bug in your ROM. This is going to be impossible to reproduce. If you want you can switch to the Play Store version and I could maybe send you a special version that contains more logging. (Play Store version is required otherwise I can’t produce APKs that you will be able to install.)

**Comment by dadosch** [comment]
OK, this is the weirdest of all bugs I ever encountered: 

I enabled the debug mode of prosody, restarted prosody (I had restarted prosody before, this didn't solve it, restarting the phone didn't help either). Then without network I opened Conversations, went to the settings. Then I turned on the network and **it works again.** Before this, this bug happened everytime I opened Conversations with network connected, now it is gone.
